src/platform/SDL2Window.cxx

```diff
@@ -187,128 +181,110 @@ void SDLWindow::swapBuffers()
 bool SDLWindow::create(void)
 {
     int targetWidth, targetHeight;
-    getSize(targetWidth, targetHeight);
-    SDL_bool windowWasGrabbed = SDL_FALSE;
-    if (windowId != NULL)
-        windowWasGrabbed = SDL_GetWindowGrab(windowId);
-
-    // if we have an existing identical window, go no further
-    if (windowId != NULL)
+    // If fullscreen, the target size will be that of the selected fullscreen resolution
+    if (fullScreen)
+        const_cast<SDLDisplay *>((const SDLDisplay *)getDisplay())->getWindowSize(targetWidth, targetHeight);
+    // Otherwise, it will be the base size, which controls the windowed size
+    else
     {
-        int currentWidth, currentHeight;
-        SDL_GetWindowSize(windowId, &currentWidth, &currentHeight);
-
-        Uint32 priorWindowFlags = SDL_GetWindowFlags(windowId);
-        if (fullScreen == (priorWindowFlags & SDL_WINDOW_FULLSCREEN) &&
-                targetWidth == currentWidth && targetHeight == currentHeight)
-            return true;
+        targetWidth = base_width;
+        targetHeight = base_height;
     }
 
-    // destroy the pre-existing window if it exists
-    if (windowId != NULL)
+    // Create the SDL window if it doesn't already exist
+    if (!windowId)
```

Comment:
I don't know if this is just a personal preference or if there is research to back this position, but I prefer the explicit test that was there before (updated for modern C++)
```suggestion
    if (windowId != nullptr)
```

Reply:
The opposite, I think, since the purpose of the comparison is now to create a new window versus destroying an existing one.

Reply:
Yes, of course. Hence the improved maintainability of the explicit logic test.